src/StackExchange.Redis/RedisDatabase.cs

```diff
@@ -3980,6 +4014,74 @@ protected override RedisValue[] Parse(in RawResult result, out int count)
             }
         }
 
+        private static Message CreateSortedSetRangeStoreMessage(
+            int db,
+            CommandFlags flags,
+            RedisKey sourceKey,
+            RedisKey destinationKey,
+            RedisValue start,
+            RedisValue stop,
+            SortedSetOrder sortedSetOrder,
+            Order order,
+            Exclude exclude,
+            long skip,
+            long? take)
+        {
+            if (sortedSetOrder == SortedSetOrder.ByRank)
+            {
+                if (take > 0)
+                {
+                    throw new ArgumentException("take argument is not valid when sortedSetOrder is ByRank you may want to try setting the SortedSetOrder to ByLex or ByScore", nameof(take));
+                }
+
+                if (exclude != Exclude.None)
+                {
+                    throw new ArgumentException("exclude argument is not valid when sortedSetOrder is ByRank, you may want to try setting the sortedSetOrder to ByLex or ByScore", nameof(exclude));
+                }
+
+                if (order == Order.Ascending)
+                {
+                    return Message.Create(db, flags, RedisCommand.ZRANGESTORE, destinationKey, sourceKey, start, stop);
+                }
+                else
+                {
+                    return Message.Create(db, flags, RedisCommand.ZRANGESTORE, destinationKey, sourceKey, start, stop, RedisLiterals.REV);
+                }
```

Comment:
TODO: let's make this a switch expression with a `_ =>` argument exception for order. (can do tonight)